packages/kbn-config-schema/src/types/object_type.ts

```diff
@@ -83,14 +91,18 @@ export class ObjectType<P extends Props = any> extends Type<ObjectResultType<P>>
     for (const [key, value] of Object.entries(props)) {
       schemaKeys[key] = value.getSchema();
     }
-    const schema = internals
+    let schema = internals
       .object()
       .keys(schemaKeys)
       .default()
       .optional()
       .unknown(unknowns === 'allow')
       .options({ stripUnknown: { objects: unknowns === 'ignore' } });
 
+    if (options.meta?.id) {
+      schema = schema.id(options.meta.id);
```

Comment:
Note:

* Refs are only supported for object types
* Since using the `joi` built in `.id()` mechanism we can actually delete quite a bit of code from the downstream OAS generation